docs/root/intro/arch_overview/upstream/service_discovery.rst

```diff
@@ -114,11 +114,28 @@ preferred service discovery mechanism for a few reasons:
   load balancing weight, canary status, zone, etc. These additional attributes are used globally
   by the Envoy mesh during load balancing, statistic gathering, etc.
 
+EDS can also be used with the :ref:`Locality Endpoint Discovery Service (LEDS)
+<arch_overview_dynamic_config_leds>`. In this case EDS will be used to fetch the cluster's
+locality-level configuration (e.g., load-balancing policy), and per-locality dynamic endpoint-level
+configuration (e.g., endpoint address, endpoint metadata) will be served via LEDS.
+
 The Envoy project provides reference gRPC implementations of EDS and
 :ref:`other discovery services <arch_overview_dynamic_config>`
 in both `Java <https://github.com/envoyproxy/java-control-plane>`_
 and `Go <https://github.com/envoyproxy/go-control-plane>`_.
 
+.. _arch_overview_endpoint_discovery_types_leds:
+
+Locality-Endpoint discovery service (LEDS)
+""""""""""""""""""""""""""""""""""""""""""
+
+The *locality-endpoint discovery service* is a :ref:`xDS management server based on gRPC in incremental
```

Comment:
Nit: is this a discovery service for locality endpoints, or the endpoint discovery service for localities? :)